addons/hdfs-snapshot-mirroring/src/main/java/org/apache/falcon/snapshots/retention/HdfsSnapshotEvictor.java

```diff
@@ -0,0 +1,208 @@
+            String snapshotDir = dirName + Path.SEPARATOR + HdfsSnapshotUtil.SNAPSHOT_DIR_PREFIX + Path.SEPARATOR;
+            FileStatus[] snapshots = fs.listStatus(new Path(snapshotDir));
+            if (snapshots.length <= numSnapshots) {
+                // no eviction needed
+                return;
+            }
+
+            // Sort by last modified time, ascending order.
+            Arrays.sort(snapshots, new Comparator<FileStatus>() {
+                @Override
+                public int compare(FileStatus f1, FileStatus f2) {
+                    return Long.compare(f1.getModificationTime(), f2.getModificationTime());
+                }
+            });
+
+            for (int i = 0; i < (snapshots.length - numSnapshots); i++) {
+                // delete if older than ageLimit while retaining numSnapshots
+                if (snapshots[i].getModificationTime() < evictionTime) {
+                    fs.deleteSnapshot(new Path(dirName), snapshots[i].getPath().getName());
+                }
+            }
+
+        } catch (ELException ele) {
+            LOG.warn("Unable to parse retention age limit {} {}", ageLimit, ele.getMessage());
+            throw new FalconException("Unable to parse retention age limit " + ageLimit, ele);
+        } catch (IOException ioe) {
+            LOG.warn("Unable to evict snapshots from dir {} {}", dirName, ioe);
+            throw new FalconException("Unable to evict snapshots from dir " + dirName, ioe);
+        }
+
+    }
+
+    private CommandLine getCommand(String[] args) throws org.apache.commons.cli.ParseException {
+        Options options = new Options();
+
+        Option opt = new Option(HdfsSnapshotMirrorProperties.SOURCE_NN.getName(), true, "Source Cluster");
+        opt.setRequired(true);
+        options.addOption(opt);
+        opt = new Option(HdfsSnapshotMirrorProperties.SOURCE_EXEC_URL.getName(),
+                true, "Replication instance job Exec Url");
+        opt.setRequired(true);
+        options.addOption(opt);
+        opt = new Option(HdfsSnapshotMirrorProperties.SOURCE_NN_KERBEROS_PRINCIPAL.getName(),
+                true, "Replication instance job NN Kerberos Principal");
+        opt.setRequired(false);
+        options.addOption(opt);
+        opt = new Option(HdfsSnapshotMirrorProperties.SOURCE_SNAPSHOT_DIR.getName(),
+                true, "Source snapshot-able dir to replicate");
+        opt.setRequired(true);
+        options.addOption(opt);
+
+        opt = new Option(HdfsSnapshotMirrorProperties.TARGET_NN.getName(), true, "Target Cluster");
+        opt.setRequired(true);
+        options.addOption(opt);
+        opt = new Option(HdfsSnapshotMirrorProperties.TARGET_SNAPSHOT_DIR.getName(),
+                true, "Target snapshot-able dir to replicate");
+        opt.setRequired(true);
+        options.addOption(opt);
+        opt = new Option(HdfsSnapshotMirrorProperties.TARGET_EXEC_URL.getName(),
```

Comment:
We don't need this - we should cleanup these options later unless we want to enhance the code to make use of this.